### PR TITLE
Plug in the directory-api-client

### PR DESCRIPTION
**Pull request by richtier**
note we will now need to set env variables for `API_CLIENT_API_KEY` and `API_CLIENT_BASE_URL`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,8 @@ Requires all host environment variables to be set.
 | ------------- | ------------- |
 | DIRECTORY_UI_SECRET_KEY | SECRET_KEY |
 | DIRECTORY_UI_PORT | PORT |
+| DIRECTORY_UI_API_CLIENT_API_KEY | API_CLIENT_API_KEY |
```
Comment (konradko) [inline README.md]: this replaces ``UI_SECRET``

File: README.md [hunk 2]
```diff
@@ -40,6 +40,8 @@ Requires all host environment variables to be set.
...
+| DIRECTORY_UI_API_CLIENT_BASE_URL | API_CLIENT_API_KEY |
```
Comment (konradko) [inline README.md]: ```| DIRECTORY_UI_API_CLIENT_BASE_URL | API_CLIENT_BASE_URL |```

File: requirements.txt [hunk 3]
```diff
@@ -6,3 +6,4 @@ raven==5.19.0
 requests==2.10.0
 whitenoise==3.1
 git+https://github.com/uktrade/directory-validators.git@0.0.2#egg=directory_validators
+git+https://github.com/uktrade/directory-api-client.git@0.0.2#egg=directory_api_client
```
Comment (konradko) [inline requirements.txt]: doesn't this need ``-e `` in front?

File: user/tests/test_views.py [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
+
 from django.core.urlresolvers import reverse
 
-from enrolment.clients.directory_api import api_client
 from user.views import UserProfileDetailView, UserProfileEditView
 from user import forms
 
 
-@mock.patch.object(api_client.user, 'retrieve_profile')
+@mock.patch.object(UserAPIClient, 'retrieve_profile')
```
Comment (konradko) [inline user/tests/test_views.py]: in the views you use ``DirectoryAPIClient`` and then components of it are mocked in the tests, quite implicit

File: requirements.txt (hunk 3 above)
Comment (richtier) [inline requirements.txt]: worked for me, so I don't think so.

Comment (konradko) [inline requirements.txt]: cool

File: user/tests/test_views.py (hunk 4 above)
Comment (konradko) [inline user/tests/test_views.py]: I would suggest mocking ``DirectoryAPIClient`` or ``directory_api_client.client``. Maybe not worth it, fine to keep as it is

**Pull request merged**